README.rst

```diff
@@ -84,137 +88,171 @@ Here's a short example of extracting citations and their metadata from text::
+
+However, recall that if we had actually overridden the resolution function before, we would have more information to possibly combine with our annotations. For example,
+let's assume that our custom :code:`get_case_by_citation()` function returned an object representing :code:`Marbury v. Madison` that contained a specific URL for that case.
+This is how we might use that information::
+
+    annotations2 = []
+    for resource, citations in resolutions.items():
+        if isinstance(resource, SomeCustomObject):
+            # add hypothetical link to case we were able to resolve using our custom function above
             url = resource.frontend_url
         else:
-            # add link to case we weren't able to resolve:
-            url = f"/unknown_cite?cite={resource.citation.matched_text()}"
-        for cite in cites:
-            annotations.append((cite.span(), f"<a href='{url}'>", f"</a>"))
+            # add generic link to all the other citations
+            url = f"/some_api?cite={resource.citation.matched_text()}"
+        for citation in citations:
+            annotations2.append((citation.span(), f"<a href='{url}'>", f"</a>"))
 
-Now we have annotations ready to add to :code:`clean_text`, but we actually want to insert them into our original
-:code:`text` variable with HTML formatting. We can pass :code:`source_text=text` into :code:`annotate()` to have the
-annotation positions adjusted and inserted into :code:`text` using the diff-match-patch library::
+    # annotations2 (pseudo):
+    # [
+    #   ((0, 11), "<a href=ACTUAL_URL_TO_CASE>", '</a>'),
+    #   ((13, 25), "<a href=ACTUAL_URL_TO_CASE>", '</a>'),
+    #   ((27, 30), "<a href=ACTUAL_URL_TO_CASE>", '</a>'),
+    #   ((31, 57), "<a href='/some_api?cite=Mass. Gen. Laws ch. 1, § 2'>", '</a>')]
+    # ]
 
-    annotated_text = annotate(cleaned_text, annotations, source_text=text)
+In either case, we now have annotations ready to add to the :code:`clean_text`, but we actually want to insert them into our original
+:code:`text` variable with the HTML formatting. To do this, we can pass :code:`source_text=text` into :code:`annotate_citations()`, which will
+intelligently adjust the annotation positions using the diff-match-patch library::
+
+    annotated_text = annotate_citations(cleaned_text, annotations, source_text=text)
 
     # annotated_text:
     # """
-    #     <p><a href='/us/1/2/'>1 <i>U.S.</i> 2</a>, <a href='/us/1/2/'>1 S.Ct. 2</a>.<p>
-    #     <p><a href='/us/1/2/'>Id.</a></p>
-    #     <p><a href='/unknown_cite?cite=Mass. Gen. Laws ch. 1, § 2'>Mass. Gen.    Laws ch. 1, § 2</a>.</p>
+    #    <p><a href='/some_api?cite=5 U. S. 137'>5 <i>U. S.</i> 137</a>, <a href='/some_api?cite=1 Cranch 137'>1 Cranch 137</a>.<p>
+    #    <p><a href='/some_api?cite=1 Cranch 137'>Id.</a></p>
+    #    <p><a href='/some_api?cite=Mass. Gen. Laws ch. 1, § 2'>Mass. Gen.    Laws ch. 1, § 2</a>.</p>
     # """
 
 Ta da!
 
-Getting Citations
+Documentation
 =================
```

Comment:
This needs to be the same length as the title above it to work properly.

Reply:
Actually, I think it's fine as long as it's *at least as long as* the title, but I've changed it to be consistent.

Reply:
Weird! Well, thanks for tweaking anyway.